homeassistant/helpers/template_entity.py

```diff
@@ -597,3 +598,34 @@ def _render_templates(self, variables: dict[str, Any]) -> None:
                 "Error rendering %s template for %s: %s", key, self.entity_id, err
             )
             self._rendered = self._static_rendered
+
+
+class ManualTriggerEntity(TriggerBaseEntity):
+    """Template entity based on manual trigger data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: dict,
+    ) -> None:
+        """Initialize the entity."""
+        TriggerBaseEntity.__init__(self, hass, config)
+
+    async def async_added_to_hass(self) -> None:
+        """Handle being added to Home Assistant."""
+        await TriggerBaseEntity.async_added_to_hass(self)
+
+    @callback
+    def _process_manual_data(self, value: str | None = None) -> None:
+        """Process new data manually."""
```

Comment:
Explain in the docstring when an implementing class should call this.

Reply:
Text expanded but is it explanatory enough?